packages/@react-aria/collections/src/Document.ts

```diff
@@ -341,6 +356,7 @@ export class ElementNode<T> extends BaseNode<T> {
       node.colSpan = props.colSpan;
     }
 
+    // TODO: still need this, see above comment
```

Comment:
why the TODO? these seem like assertions of facts/explanations, is there a question here or can we keep the general gist of the explanation as a comment?

Reply:
yeah, they are temporary notes that I'll refine/get rid of once the approach settles. We had hoped that we could get rid of this `hasSetProps` here but in practice it broke the node ids/refs as noted. Leaving them here in case the flow seems off and we want to change some more things